### DISCUSSION TITLE
TIA606C automatic lable generation for cables

### DISCUSSION THREAD

**Discussion by CupcakeMunster** [netbox-community/netbox#10962] [Ideas]
I was just thinking that rather than labelling all my cabling myself, I could gather the data from a netbox csv to create labels that are TIA606 compliant.  Could this be something easy to implement with netbox itself? 

If I had a patching cabinet then every cable I add in netbox could have a checkbox for auto-TIA606C label generation. Thoughts?

![TIA606CLinkIdentifierTerminatedSSExample](https://user-images.githubusercontent.com/26608204/202558216-f725e0f4-d6ab-4063-91b8-1eb5dccc6092.jpg)

**Comment by iunderwood** [comment]
That's not a bad way to go for rear cables, but I find that it's way too cumbersome for front ports or anything subject to changes.

The last 3 places I have worked have kept to a relatively simple sequence on each cable.  The latest iteration is the FY of purchase, length in feet, and a sequence number.  For example, 23-5-1314.  Netbox tracks the cable and it's easy to audit each end and patch panel.

**Reply by CupcakeMunster** [reply]
I agree about how cumbersome it is. It could be an option that's enabled on rear ports and disabled on front ports depending on what the user wants. I guess I'm figuring out how plugins work. lol

**Comment by deku-m** [comment]
I know that someone had the intention of creating a plugin for it to automate the cable labeling:
But didnt start yet i think.

https://github.com/jsenecal/netbox-auto-cable-label

**Reply by CupcakeMunster** [reply]
I'm going to take a crack at plugins and see how far I can get. Don't hold your breath though.

**Reply by eronlloyd** [reply]
Awesome. I look forward to seeing how you do!

**Reply by jsenecal** [reply]
I did now, fwiw :) its released. I think that with some Jinja-Fu you could end-up with TIA606C cable labels :)
